src/Runner.Listener/MessageListener.cs

```diff
@@ -319,7 +319,8 @@ private TaskAgentMessage DecryptMessage(TaskAgentMessage message)
                 var keyManager = HostContext.GetService<IRSAKeyManager>();
                 using (var rsa = keyManager.GetKey())
                 {
-                    return aes.CreateDecryptor(rsa.Decrypt(_session.EncryptionKey.Value, RSAEncryptionPadding.OaepSHA1), message.IV);
+                    var padding = _settings.RequireFipsCryptography ? RSAEncryptionPadding.OaepSHA256 : RSAEncryptionPadding.OaepSHA1;
```

Comment:
this is the tricky part that makes you do a runner setting, i think.

Reply:
Yea it was find a single best place for this. Putting it in `credentialData` makes sense too, I will try that

Reply:
I opted to add an extra property to the session since thats where the session key is, and the way that key is encrypted wont change for the life of a session (and its key)